fitbenchmarking/controllers/mantid_controller.py

```diff
@@ -107,6 +109,19 @@ def __init__(self, cost_func):
                                 for i in range(1, self._multi_fit))
                 function_def += 'ties=({})'.format(ties)
 
+            # Add constraints if parameter bounds are set
+            if self.value_ranges is not None and self._multi_fit:
+                constraints = ','.join('{0} < f{1}.{2} < {3}'.format(
+                    self.value_ranges[i][0], j, p, self.value_ranges[i][1])
+                    for i, p in enumerate(self._param_names)
+                    for j in range(0, self._multi_fit))
+                function_def += '; constraints=({})'.format(constraints)
+            elif self.value_ranges is not None:
+                constraints = ','.join('{0} < f0.{1} < {2}'.format(
+                    self.value_ranges[i][0], p, self.value_ranges[i][1])
+                    for i, p in enumerate(self._param_names))
```

Comment:
When testing with:
```
function = 'name=FlatBackground,A0=0.05;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=0.1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=5,Phi=0'
parameter_ranges = {'f0_A0': [0.045, 100]}
```
I get:
```
########################################################
 Running data from: Muon_67068_coadd_67108_long.txt 1/4
########################################################
    Starting value: 1/1
        Software: MANTID
            Minimizer: Levenberg-Marquardt
                Jacobian: scipy 2-point
Invalid value for property Function (Function) from string "name=FlatBackground,A0=0.05;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=0.1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=5,Phi=0; constraints=(0.045 < f0.f0_A0 < 100.0,-inf < f0.f1_A < inf,-inf < f0.f1_Lambda < inf,-inf < f0.f1_Frequency < inf,-inf < f0.f1_Phi < inf,-inf < f0.f2_A < inf,-inf < f0.f2_Lambda < inf,-inf < f0.f2_Frequency < inf,-inf < f0.f2_Phi < inf,-inf < f0.f3_A < inf,-inf < f0.f3_Lambda < inf,-inf < f0.f3_Frequency < inf,-inf < f0.f3_Phi < inf)": ParamFunction FlatBackground does not have parameter (f0_A0).
            Minimizer: Levenberg-MarquardtMD
                Jacobian: scipy 2-point
Invalid value for property Function (Function) from string "name=FlatBackground,A0=0.05;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=0.1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=5,Phi=0; constraints=(0.045 < f0.f0_A0 < 100.0,-inf < f0.f1_A < inf,-inf < f0.f1_Lambda < inf,-inf < f0.f1_Frequency < inf,-inf < f0.f1_Phi < inf,-inf < f0.f2_A < inf,-inf < f0.f2_Lambda < inf,-inf < f0.f2_Frequency < inf,-inf < f0.f2_Phi < inf,-inf < f0.f3_A < inf,-inf < f0.f3_Lambda < inf,-inf < f0.f3_Frequency < inf,-inf < f0.f3_Phi < inf)": ParamFunction FlatBackground does not have parameter (f0_A0).
            Minimizer: BFGS
                Jacobian: scipy 2-point
Invalid value for property Function (Function) from string "name=FlatBackground,A0=0.05;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=0.1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=5,Phi=0; constraints=(0.045 < f0.f0_A0 < 100.0,-inf < f0.f1_A < inf,-inf < f0.f1_Lambda < inf,-inf < f0.f1_Frequency < inf,-inf < f0.f1_Phi < inf,-inf < f0.f2_A < inf,-inf < f0.f2_Lambda < inf,-inf < f0.f2_Frequency < inf,-inf < f0.f2_Phi < inf,-inf < f0.f3_A < inf,-inf < f0.f3_Lambda < inf,-inf < f0.f3_Frequency < inf,-inf < f0.f3_Phi < inf)": ParamFunction FlatBackground does not have parameter (f0_A0).
            Minimizer: Conjugate gradient (Fletcher-Reeves imp.)
                Jacobian: scipy 2-point
Invalid value for property Function (Function) from string "name=FlatBackground,A0=0.05;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=0.1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=5,Phi=0; constraints=(0.045 < f0.f0_A0 < 100.0,-inf < f0.f1_A < inf,-inf < f0.f1_Lambda < inf,-inf < f0.f1_Frequency < inf,-inf < f0.f1_Phi < inf,-inf < f0.f2_A < inf,-inf < f0.f2_Lambda < inf,-inf < f0.f2_Frequency < inf,-inf < f0.f2_Phi < inf,-inf < f0.f3_A < inf,-inf < f0.f3_Lambda < inf,-inf < f0.f3_Frequency < inf,-inf < f0.f3_Phi < inf)": ParamFunction FlatBackground does not have parameter (f0_A0).
            Minimizer: Conjugate gradient (Polak-Ribiere imp.)
                Jacobian: scipy 2-point
Invalid value for property Function (Function) from string "name=FlatBackground,A0=0.05;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=0.1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=5,Phi=0; constraints=(0.045 < f0.f0_A0 < 100.0,-inf < f0.f1_A < inf,-inf < f0.f1_Lambda < inf,-inf < f0.f1_Frequency < inf,-inf < f0.f1_Phi < inf,-inf < f0.f2_A < inf,-inf < f0.f2_Lambda < inf,-inf < f0.f2_Frequency < inf,-inf < f0.f2_Phi < inf,-inf < f0.f3_A < inf,-inf < f0.f3_Lambda < inf,-inf < f0.f3_Frequency < inf,-inf < f0.f3_Phi < inf)": ParamFunction FlatBackground does not have parameter (f0_A0).
            Minimizer: Damped GaussNewton
                Jacobian: scipy 2-point
Invalid value for property Function (Function) from string "name=FlatBackground,A0=0.05;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=0.1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=5,Phi=0; constraints=(0.045 < f0.f0_A0 < 100.0,-inf < f0.f1_A < inf,-inf < f0.f1_Lambda < inf,-inf < f0.f1_Frequency < inf,-inf < f0.f1_Phi < inf,-inf < f0.f2_A < inf,-inf < f0.f2_Lambda < inf,-inf < f0.f2_Frequency < inf,-inf < f0.f2_Phi < inf,-inf < f0.f3_A < inf,-inf < f0.f3_Lambda < inf,-inf < f0.f3_Frequency < inf,-inf < f0.f3_Phi < inf)": ParamFunction FlatBackground does not have parameter (f0_A0).
            Minimizer: Simplex
Invalid value for property Function (Function) from string "name=FlatBackground,A0=0.05;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=0.1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=5,Phi=0; constraints=(0.045 < f0.f0_A0 < 100.0,-inf < f0.f1_A < inf,-inf < f0.f1_Lambda < inf,-inf < f0.f1_Frequency < inf,-inf < f0.f1_Phi < inf,-inf < f0.f2_A < inf,-inf < f0.f2_Lambda < inf,-inf < f0.f2_Frequency < inf,-inf < f0.f2_Phi < inf,-inf < f0.f3_A < inf,-inf < f0.f3_Lambda < inf,-inf < f0.f3_Frequency < inf,-inf < f0.f3_Phi < inf)": ParamFunction FlatBackground does not have parameter (f0_A0).
            Minimizer: SteepestDescent
                Jacobian: scipy 2-point
Invalid value for property Function (Function) from string "name=FlatBackground,A0=0.05;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=0.1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=1,Phi=0;name=ExpDecayOsc,A=0.2,Lambda=0.2,Frequency=5,Phi=0; constraints=(0.045 < f0.f0_A0 < 100.0,-inf < f0.f1_A < inf,-inf < f0.f1_Lambda < inf,-inf < f0.f1_Frequency < inf,-inf < f0.f1_Phi < inf,-inf < f0.f2_A < inf,-inf < f0.f2_Lambda < inf,-inf < f0.f2_Frequency < inf,-inf < f0.f2_Phi < inf,-inf < f0.f3_A < inf,-inf < f0.f3_Lambda < inf,-inf < f0.f3_Frequency < inf,-inf < f0.f3_Phi < inf)": ParamFunction FlatBackground does not have parameter (f0_A0).
```

It looks like it's setting the right bounds on the string, but I guess it's looking for `f0.A0` rather than `f0.f0_A0`